codec/src/test/java/io/netty/handler/codec/compression/Lz4FrameEncoderTest.java

```diff
@@ -106,6 +112,20 @@ private void testAllocateBuffer(int bufSize, boolean isDirect) {
         }
     }
 
+    @Test (expected = EncoderException.class)
+    public void testAllocateDirectBuffer_ExceedMaxEncodeSize() {
+        final int maxEncodeSize = 1024;
+        Checksum checksum = XXHashFactory.fastestInstance().newStreamingHash32(DEFAULT_SEED).asChecksum();
+        Lz4FrameEncoder encoder = new Lz4FrameEncoder(LZ4Factory.fastestInstance(), true,
+                                                      Lz4Constants.DEFAULT_BLOCK_SIZE,
+                                                      checksum,
+                                                      maxEncodeSize);
+        int inputBufferSize = maxEncodeSize * 10;
+        ByteBuf buf = PooledByteBufAllocator.DEFAULT.buffer(inputBufferSize, inputBufferSize);
+        buf.writerIndex(inputBufferSize);
+        encoder.allocateBuffer(null, buf, false);
```

Comment:
I believe the `channel.finish()` and `channel.release*()` methods will already release the buffer.